src/Page/Food/Builder.elm

```diff
@@ -590,50 +591,80 @@ updateIngredientFormView { excluded, db, ingredient, impact, transportImpact } =
-                                            if checked then
-                                                Ingredient.ByPlane
-
-                                            else
-                                                Ingredient.NoPlane
-                                    }
-                            )
+                        , checked isByPlane
+                        , onInput <| always (event { ingredientQuery | planeTransport = Ingredient.ByPlane })
                         ]
                         []
-                    , text " par avion)"
+                    , Icon.plane
                     ]
+                , label [ class "BoatSelector" ]
+                    [ input
+                        [ type_ "radio"
+                        , attribute "role" "switch"
+                        , checked <| not isByPlane
+                        , onInput <| always (event { ingredientQuery | planeTransport = Ingredient.NoPlane })
+                        ]
+                        []
+                    , Icon.boat
+                    ]
+                , if isByPlane then
+                    span [ class "ps-1 align-items-center gap-1", title "Tranport aérien" ]
+                        [ Format.km air ]
 
-              else
-                text ""
+                  else
+                    span [ class "ps-1 align-items-center gap-1", title "Tranport maritime" ]
+                        [ Format.km sea ]
+                ]
+            , TransportView.entry road Icon.bus "Transport routier"
             ]
-        , ingredient
-            |> Recipe.computeIngredientTransport db
-            |> TransportView.viewDetails
-                { fullWidth = False
-                , airTransportLabel = Nothing
-                , seaTransportLabel = Nothing
-                , roadTransportLabel = Nothing
-                }
-            |> span [ class "text-muted d-flex fs-7 gap-3 justify-content-left IngredientTransportDistances" ]
-        , span [ class "text-muted text-end IngredientTransportImpact fs-7" ]
-            [ transportImpact ]
-        ]
-    ]
+
+         else
+            ingredient
+                |> Recipe.computeIngredientTransport db
+                |> (\{ air, sea, road } ->
+                        [ { distance = air, icon = Icon.plane, label = "Transport aérien" }
+                        , { distance = sea, icon = Icon.boat, label = "Transport maritime" }
+                        , { distance = road, icon = Icon.bus, label = "Transport routier" }
+                        ]
+                            |> List.filterMap
+                                (\{ distance, icon, label } ->
+                                    if Length.inKilometers distance /= 0 then
+                                        Just <| TransportView.entry distance icon label
+
+                                    else
+                                        Nothing
+                                )
```

Comment:
Nit: Wouldn't this a good use case for adapting/extending `TransportView.viewDetails`?

Reply:
good idea ;)